specifyweb/frontend/js_src/lib/components/Login/index.tsx

```diff
@@ -22,9 +24,40 @@ import type { OicProvider } from './OicLogin';
 import { OicLogin } from './OicLogin';
 
 export function Login(): JSX.Element {
+  const [institutions] = useAsyncState(
+    React.useCallback(
+      async () =>
+        ajax<RA<string>>(`/api/specify/institutions/`, {
+          method: 'GET',
+          headers: {
+            Accept: 'application/json',
+          },
+          errorMode: 'silent',
+        })
+          .then(({ data }) => {
+            if (data.length === 0) {
+              console.warn("No institutions found.");
+            }
+            return data;
+          })
+          .catch((error) => {
+            console.error("Failed to fetch institutions:", error);
+            return undefined;
+          }),
+      []
+    ),
+    true
+  );
+
   return React.useMemo(() => {
     const nextUrl = parseDjangoDump<string>('next-url') ?? '/specify/';
     const providers = parseDjangoDump<RA<OicProvider>>('providers') ?? [];
+
+    if (institutions?.length === 0 || institutions === undefined) {
+      // Display here the new setup pages 
+      return <p>Welcome! No institutions are available at the moment.</p>;
+    }
```

Comment:
I would suggest having this block outside the memoized component that is returned. Looks like the memo is for identifying which login provider needs to be returned so I think this can go outside and would also eliminate the need for institution as a dependency